app/src/screens/profile.screen.tsx

```diff
@@ -0,0 +1,380 @@
+          <Tabs w={'100%'} variant='enclosed'>
+            <TabList>
+              <Tab>Personal Information</Tab>
+              <Tab>Actions</Tab>
+            </TabList>
+            <TabPanels>
+              <TabPanel>
+                <div className='flex justify-between flex-wrap gap-5'>
+                  <CustomInputBox label={'Email'} value={user?.email || ''} />
+                  <CustomInputBox
+                    label={'Display Name'}
+                    value={user?.displayname || ''}
+                  />
+                  <CustomInputBox
+                    label={`Last password change (${relativeTimeLabel})`}
+                    value={
+                      user?.passChangedAt
+                        ? typeof user.passChangedAt === 'string'
+                          ? new Date(user.passChangedAt).toLocaleTimeString(
+                              'en-US',
+                              {
+                                hour12: false,
+                              },
+                            ) +
+                            ' ' +
+                            new Date(user.passChangedAt)
+                              .toLocaleDateString('en-US')
+                              .split('/')
+                              .reverse()
+                              .join('/')
+                          : user.passChangedAt.toLocaleTimeString('en-US', {
+                              hour12: false,
+                            }) +
+                            ' ' +
+                            user.passChangedAt
+                              .toLocaleDateString('en-US')
+                              .split('/')
+                              .reverse()
+                              .join('/')
+                        : ''
+                    }
+                  />
+                </div>
+              </TabPanel>
+              <TabPanel>
+                <div className='flex flex-col justify-start gap-5'>
+                  <ChangePasswordModal />
+                  <ChangeDisplayNameModal />
+                </div>
+              </TabPanel>
+            </TabPanels>
+          </Tabs>
+        </div>
+      </div>
+    );
+  };
+
+  return (
+    <div className='flex flex-col h-full w-full justify-center items-center'>
+      <div className='flex flex-col h-full w-2/3'>
+        <div className='flex w-full justify-start pb-4 items-center'>
+          <IconButton aria-label='Back icon' onClick={() => navigateBack()}>
+            <ArrowBackIcon boxSize={5} />
+          </IconButton>
+          <div className='flex flex-col leading-none w-full justify-center items-center px-[40px] mb-5 h-[40px]'>
+            <Text
+              fontSize={{ base: 'x-large', md: 'xx-large' }}
+              fontWeight={'bold'}
+            >
+              Profile
+            </Text>
+          </div>
+        </div>
+        {user ? <ProfileCard /> : <Spinner />}
+      </div>
```

Comment:
In the case where `user` data is empty this will always show a spinner, I think that the hook has an isLoading property

Reply:
if user data is empty, it means there is no logged in user, so it will take to the login page, in the other case, it will wait till data is loaded